crates/matrix-sdk-ui/src/timeline/pagination.rs

```diff
@@ -32,8 +32,21 @@ impl super::Timeline {
     ///
     /// Returns whether we hit the start of the timeline.
     #[instrument(skip_all, fields(room_id = ?self.room().room_id()))]
-    pub async fn paginate_backwards(&self, num_events: u16) -> Result<bool, Error> {
+    pub async fn paginate_backwards(&self, mut num_events: u16) -> Result<bool, Error> {
         if self.controller.is_live().await {
+            match self.controller.live_lazy_paginate_backwards(num_events).await {
+                Some(needed_num_events) => {
```

Comment:
Would it maybe be better to have `live_lazy_paginate_backwards` return the right type?

Reply:
What do you mean by the _right type_?